rules/import-path-order.js

```diff
@@ -0,0 +1,473 @@
+'use strict';
+const isBuiltin = require('is-builtin-module');
+const getDocsUrl = require('./utils/get-docs-url');
+
+const MESSAGE_ID_BLANKLINES = 'import-path-blanklines';
+const MESSAGE_ID_DEPTH = 'import-path-order-depth';
+const MESSAGE_ID_GROUP = 'import-path-order-group';
+const MESSAGE_ID_ORDER = 'import-path-order';
+
+const GROUP_BUILTIN = 1;
+const GROUP_ABSOLUTE = 2;
+const GROUP_PARENT = 3;
+const GROUP_SIBLING = 4;
+
+const GROUP_NAMES = {
+	[GROUP_BUILTIN]: 'Built-in',
+	[GROUP_ABSOLUTE]: 'Absolute',
+	[GROUP_PARENT]: 'Relative',
+	[GROUP_SIBLING]: 'Sibling'
+};
+
+const COMPARATOR_CASE_SENSITIVE = 'case-sensitive';
+const COMPARATOR_CASE_INSENSITIVE = 'case-insensitive';
+const COMPARATOR_CASE_PARTS = 'parts';
+const COMPARATOR_CASE_OFF = 'off';
+
+function getOrder(name, partsRegex) {
+	const parts = {};
+	for (const regex of partsRegex) {
+		parts[regex] = name.split(new RegExp(regex));
+	}
+
+	if (isBuiltin(name)) {
+		return {
+			name,
+			group: GROUP_BUILTIN,
+			depth: 0,
+			parts
+		};
+	}
+
+	if (name.match(/^\.\//)) {
+		return {
+			name,
+			group: GROUP_SIBLING,
+			depth: 0,
+			parts
+		};
+	}
+
+	const relative = name.match(/^((\.\.\/)+)/);
+	if (relative) {
+		return {
+			name,
+			group: GROUP_PARENT,
+			depth: relative[1].split('..').length,
+			parts
+		};
+	}
+
+	return {
+		name,
+		group: GROUP_ABSOLUTE,
+		depth: 0,
+		parts
+	};
+}
+
+function getInvalidBlankLinesReport(nodePrev, nodeNext, context) {
+	if (nodePrev === null) {
+		return null;
```

Comment:
Can you use `undefined` everywhere instead of `null`?

https://github.com/sindresorhus/meta/issues/7

Reply:
Oh, yeah, I keep forgetting that you prefer it that way.